sdk/core/core-auth/src/azureNamedKeyCredential.ts

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * Represents a credential defined by a static API name and key.
+ */
+export interface NamedKeyCredential {
+  /**
+   * The value of the API key represented as a string
+   */
+  readonly key: string;
+  /**
+   * The value of the API name represented as a string.
+   */
+  readonly name: string;
+}
+
+/**
+ * A static name/key-based credential that supports updating
+ * the underlying name and key values.
+ */
+export class AzureNamedKeyCredential implements NamedKeyCredential {
+  private _key: string;
+  private _name: string;
+
+  /**
+   * The value of the key to be used in authentication.
+   */
+  public get key(): string {
```

Comment:
any particular reason these are getters instead of normal props?

Reply:
We were wanting the user to have to call `update()` to update the key, and otherwise treat these as readonly properties. This is the pattern the other credentials in this package (e.g. `AzureKeyCredential`) follows as well.

Reply:
Yeah that makes sense